File: source/manual/ds_writing/cpp_started/DevArray.cpp
Tango::DevVarLongArray *DocDs::dev_array(const Tango::DevVarLongArray *argin)
{
	//	POGO has generated a method core with argout allocation.
	//	If you would like to use a static reference without copying,
	//	See "TANGO Device Server Programmer's Manual"
	//		(chapter x.x)
	//------------------------------------------------------------
	Tango::DevVarLongArray	*argout  = new Tango::DevVarLongArray();
		
	cout1 << "DocDs::dev_array(): entering... !" << endl;

	//	Add your own code to control device here

	long argin_length = argin->length();	
	argout->length(argin_length);
	for (int i = 0;i < argin_length;i++)
		(*argout)[i] = (*argin)[i] * 2;

	return argout;
}

File: source/development/device-api/device-server-writing/java_started/DevArray.java
public int[] dev_array(int[] argin) throws DevFailed
{
	int[]	argout = new int[argin.length];

	Util.out2.println("Entering dev_array()");

	// ---Add your Own code to control device here ---

	for (int i = 0;i < argin.length;i++)
		argout[i] = argin[i] * 2;
	return argout;
}
